test/functional/autocmd/termclose_spec.lua

```diff
@@ -20,6 +20,7 @@ describe('TermClose event', function()
   it('triggers when fast-exiting terminal job stops', function()
     command('autocmd TermClose * let g:test_termclose = 23')
     command('terminal')
+    retry(nil, nil, function() neq(0, eval('b:terminal_job_pid')) end)
     command('call jobstop(b:terminal_job_id)')
```

Comment:
I think there isn't a race here because `:terminal` command is synchronous: `b:terminal_job_pid` is guaranteed to be valid when control returns.

Reply:
Makes sense.